pkg/apis/jaegertracing/v1/jaeger_types.go

```diff
@@ -49,11 +49,33 @@ const (
 	IngressSecurityOAuthProxy IngressSecurityType = "oauth-proxy"
 )
 
+// DeploymentStrategy represents the possible values for deployment strategies
+// +k8s:openapi-gen=true
+type DeploymentStrategy string
```

Comment:
You could implement the UnmarshalJSON method to the type, to ensure the values read from JSON are converted to lowercase.